_gitbook/documenting_code/README.md

```diff
@@ -1,6 +1,112 @@
 # Documenting code
 
-Crystal documentation comments use a subset of [Markdown](https://daringfireball.net/projects/markdown/). For example:
+Crystal documentation comments use a subset of [Markdown](https://daringfireball.net/projects/markdown/). 
+
+### Instructions
+
+Documents should be positioned right above definitions of classes, modules, and methods. Leave no blanks between them.
+
+```crystal
+# A unicorn is a **legendary animal** (see the `Legendary` module) that has been
+# described since antiquity as a beast with a large, spiraling horn projecting
+# from its forhead.
+class Unicorn
+end
+
+# Bad: This comment is not attached to any class.
+ 
+class Legendary
+end
+```
+
+* Methods' documentation include Method Summary and Method Detail, while the former includes only the first line of the comments and the latter includes the entire paragraph. In short, a good documentation for a method is to:
+
+  1. State its purpose or functionality in the first line.
+  2. Supplement with details and usages after that.
+
+For instance:
+
+```crystal
+# Returns the number of horns this unicorn has
+#
+# ```
+# Unicorn.new.horns #=> 1
+# ```
+def horns
+  @horns
+end
+```
+
+* Use the third person: `Returns the number` instead of `Return the number`.
```

Comment:
The original example is `Returns the number of horns`. I think I'll use this one.